### PR TITLE
[platforms] refactor cpu code

### PR DESCRIPTION
**Pull request by youkaichao**
part of https://github.com/vllm-project/vllm/issues/9268

### CODE REVIEW

File: vllm/platforms/cpu.py [hunk 1]
```diff
@@ -18,3 +29,52 @@ def get_device_total_memory(cls, device_id: int = 0) -> int:
     @classmethod
     def inference_mode(cls):
         return torch.no_grad()
+
+    @classmethod
+    def check_and_update_config(cls, vllm_config: VllmConfig) -> None:
```
Comment (DarkLight1337) [inline vllm/platforms/cpu.py:34]: This doesn't seem to be called anywhere.

Comment (youkaichao) [reply vllm/platforms/cpu.py:34]: it is called at https://github.com/vllm-project/vllm/blob/643ecf7b11a3e74c838f438cfc1b3e59c018853b/vllm/config.py#L2323

Comment (DarkLight1337) [reply vllm/platforms/cpu.py:34]: I see, looks good then.

**Review by DarkLight1337 — APPROVED** [review]

**Comment by DarkLight1337** [comment]
Please check that CPU tests have not soft failed before merging.

**Comment by DarkLight1337** [comment]
Might need to merge the fix from #10394 first, but let's see.

**Comment by youkaichao** [comment]
> Please check that CPU tests have not soft failed before merging.

CPU tests passed.

File: vllm/platforms/cpu.py (hunk 1 above)
Comment (Xerxes-cn) [reply vllm/platforms/cpu.py:34]: 这里的写法是不是有点问题
```python
if TYPE_CHECKING:
    from vllm.config import VllmConfig

def check_and_update_config(cls, vllm_config: "VllmConfig") -> None:
    ...
```
使用引号就行了。而不是用else VllmConfig = None

Comment (youkaichao) [reply vllm/platforms/cpu.py:34]: either should work

**Pull request merged**